src/immut_helloworld.spec.ts

```diff
@@ -0,0 +1,323 @@
+import {Uint64} from "./custom_types";
+import {helloworld} from "./helloworld";
+import {HelloRequest, Person, HelloReply, PersonContactEnum} from "./immut_helloworld";
+
+describe("HelloRequest", () => {
+
+	describe("New HelloRequest", () => {
+		it("should create a new instance without throwing error", () => {
+			expect(() => new HelloRequest()).not.toThrow();
+		});
+	});
+
+	describe("with a new instance of HelloRequest", () => {
```

Comment:
* Immutable-esque behaviour; we get a new object, don't get a new object if value hasn't changed etc....
* Default values are what we expect
* Setting of `undefined` etc behaves as we expect